sycl/source/detail/graph_impl.cpp

```diff
@@ -69,6 +94,16 @@ node_ptr graph_impl::add(graph_ptr impl, T cgf,
   }
   return nodeImpl;
 }
+    
+node_ptr graph_impl::add_malloc(graph_ptr impl, 
+                                void*& ptr, size_t count, 
+                                sycl::usm::alloc kind, const std::vector<node_ptr> &dep) {
+    // TODO: Do alloc within the graph execution.
+    // For now return dummy node without any dependencies
+    node_ptr nodeImpl = std::make_shared<node_impl>(impl);
+    MAllocs.push_back(std::tie(ptr,count,kind));
```

Comment:
```suggestion
    MAllocs.push_back(usm_mem_info{ptr,count,kind});
```

Reply:
`emplace_back` would probably be preferable here to avoid copies.